src/java/fr/paris/lutece/plugins/mylutece/modules/users/service/LocalUserRoleService.java

```diff
@@ -0,0 +1,86 @@
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * License 1.0
+ */
+
+package fr.paris.lutece.plugins.mylutece.modules.users.service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import fr.paris.lutece.plugins.mylutece.modules.users.business.LocalUserRole;
+import fr.paris.lutece.plugins.mylutece.modules.users.business.LocalUserRoleHome;
+
+/**
+ * Service class for LocalUserRole
+ *
+ */
+public final class LocalUserRoleService
+{
+    public static final String BEAN_NAME = "mylutece-users.localUserRoleService";
+
+    /**
+     * Process to user assignements from Mylutece Role List
+     * 
+     * @param myLuteceRoleKeysList
+     *            The MyLutece role keys List
+     * @param nIdLocalUser
+     *            The localUser id
+     */
+    public static void doLocalUserAssignements( List<String> myLuteceRoleKeysList, int nIdLocalUser )
+    {
+
+        List<LocalUserRole> localUserRolesList = LocalUserRoleHome.getLocalUserRolesListByUserId( nIdLocalUser );
+        if ( myLuteceRoleKeysList == null )
+        {
+            LocalUserRoleHome.removeFromLocalUserRoleList( localUserRolesList );
+            return;
+        }
+        // find all localUser roles to delete.
+        List<LocalUserRole> localUsersRolesToDeleteList = localUserRolesList.stream( )
+                .filter( currentLocalUserRoleKey -> !myLuteceRoleKeysList.contains( currentLocalUserRoleKey.getRoleKey( ) ) ).collect( Collectors.toList( ) );
+        // find all myLutece roles to assign.
+        List<String> myLuteceRoleKeyToAssignList = myLuteceRoleKeysList.stream( )
+                .filter( myLuteceRoleKey -> !localUserRolesList.stream( ).anyMatch( localUserRole -> myLuteceRoleKey.equals( localUserRole.getRoleKey( ) ) ) )
+                .collect( Collectors.toList( ) );
+
+        if ( localUsersRolesToDeleteList != null )
+        {
+            LocalUserRoleHome.removeFromLocalUserRoleList( localUsersRolesToDeleteList );
+        }
+        if ( myLuteceRoleKeyToAssignList != null )
+        {
+            LocalUserRoleHome.createFromLocalUserRoleKeyList( myLuteceRoleKeyToAssignList, nIdLocalUser );
```

Comment:
this method cuold be named "createLocalUserRoles"

Reply:
done